CollAction/Helpers/ProjectNormalizeStringExtensions.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Text;
+using System.Globalization;
+using System.Text.RegularExpressions;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CollAction.Helpers
+{
+    public static class ProjectNormalizeStringExtensions
+    {
+
+        public static string ToUrlSlug(this string value){
+
+            //First to lower case
+            value = value.ToLowerInvariant();
+
+            //Replace spaces
+            value = Regex.Replace(value, @"\s", "-", RegexOptions.Compiled);
+
+            //Remove invalid chars
+            value = Regex.Replace(value, @"[^a-z0-9\s-_]", "",RegexOptions.Compiled);
+
+            //Trim dashes from end
+            value = value.Trim('-', '_');
+
+            //Replace double occurences of - or _
+            value = Regex.Replace(value, @"([-_]){2,}", "$1", RegexOptions.Compiled);
+
+            return value;
+        }
+    
+        public static string RemoveAccent(this string txt) 
+        { 
+            byte[] bytes = System.Text.Encoding.GetEncoding("Cyrillic").GetBytes(txt); 
+            return System.Text.Encoding.ASCII.GetString(bytes); 
+        }
+
+        public static string Translit(this string str)
+        {
+            string[] lat_up = {"A", "B", "V", "G", "D", "E", "Yo", "Zh", "Z", "I", "Y", "K", "L", "M", "N", "O", "P", "R", "S", "T", "U", "F", "Kh", "Ts", "Ch", "Sh", "Shch", "\"", "Y", "'", "E", "Yu", "Ya"};
```

Comment:
Make your arrays static to keep the compiler from allocating them again and again.

Reply:
Gotta say, this might be overkill (doubt we'll ever be popular in Russia), but it's also impressively complete.

Reply:
That is true, I will remove it for now to improve performance for 99.99% use cases and cleanup code, placeholder for empty strings solves problem.